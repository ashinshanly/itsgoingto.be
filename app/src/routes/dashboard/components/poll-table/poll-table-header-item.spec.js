/* global expect */
import React from 'react'
import { shallow } from 'enzyme'
import { PollTableHeaderItem } from './poll-table-header-item';

const props = {
  label: 'Column header',
  style: {},
  sortDirection: false,
  onSort: jest.fn(),
}
let wrapper

describe('(Route) dashboard', () => {
  describe('(Component) poll table header item', () => {
    beforeEach(() => {
      wrapper = shallow(<PollTableHeaderItem {...props} />)
    })

    afterEach(() => {
      jest.clearAllMocks()
    })

    describe('(Action) onClick', () => {
      describe('has onSort callback', () => {
        const onSort = jest.fn()

        beforeEach(() => {
          wrapper = shallow(<PollTableHeaderItem {...props} onSort={onSort} />)
        })

        it('calls onSort', () => {
          wrapper.find('a').simulate('click')
          expect(onSort).toHaveBeenCalled()
        })
      })
    })

    describe('(Render)', () => {
      describe('has onSort callback', () => {
        it('matches snapshot', () => {
          wrapper = shallow(<PollTableHeaderItem {...props} onSort={() => { }} />)
          expect(wrapper).toMatchSnapshot()
        })

        describe('has sortDirection asc', () => {
          it('matches snapshot', () => {
            wrapper = shallow(<PollTableHeaderItem {...props} onSort={() => { }} sortDirection='asc' />)
            expect(wrapper).toMatchSnapshot()
          })
        })

        describe('has sortDirection desc', () => {
          it('matches snapshot', () => {
            wrapper = shallow(<PollTableHeaderItem {...props} onSort={() => { }} sortDirection='desc' />)
            expect(wrapper).toMatchSnapshot()
          })
        })
      })

      describe('doesn\'t have onSort callback', () => {
        it('matches snapshot', () => {
          wrapper = shallow(<PollTableHeaderItem {...props} onSort={null} />)
          expect(wrapper).toMatchSnapshot()
        })
      })
    })
  })
})