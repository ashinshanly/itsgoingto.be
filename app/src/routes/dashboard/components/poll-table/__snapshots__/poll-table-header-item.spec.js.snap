// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) dashboard (Component) poll table header item (Render) doesn't have onSort callback matches snapshot 1`] = `
<th
  className={false}
  style={Object {}}
>
  <span>
    Column header
  </span>
</th>
`;

exports[`(Route) dashboard (Component) poll table header item (Render) has onSort callback has sortDirection asc matches snapshot 1`] = `
<th
  className="sort-active"
  style={Object {}}
>
  <a
    onClick={[Function]}
  >
    Column header
     
    <i
      className="fa fa-sort-up"
    />
  </a>
</th>
`;

exports[`(Route) dashboard (Component) poll table header item (Render) has onSort callback has sortDirection desc matches snapshot 1`] = `
<th
  className="sort-active"
  style={Object {}}
>
  <a
    onClick={[Function]}
  >
    Column header
     
    <i
      className="fa fa-sort-down"
    />
  </a>
</th>
`;

exports[`(Route) dashboard (Component) poll table header item (Render) has onSort callback matches snapshot 1`] = `
<th
  className={false}
  style={Object {}}
>
  <a
    onClick={[Function]}
  >
    Column header
     
    <i
      className="fa fa-sort"
    />
  </a>
</th>
`;
